portal/models/practitioner.py

```diff
@@ -0,0 +1,115 @@
+"""Practitioner module"""
+from cgi import escape
+from sqlalchemy import UniqueConstraint
+
+from ..database import db
+from .identifier import Identifier
+from .telecom import ContactPoint, Telecom
+
+
+class Practitioner(db.Model):
+    """Practitioner model for storing physician information"""
+    __tablename__ = 'practitioners'
+    id = db.Column(db.Integer, primary_key=True)
+    first_name = db.Column(db.String(64))
+    last_name = db.Column(db.String(64), nullable=False)
+    phone_id = db.Column(db.Integer, db.ForeignKey('contact_points.id',
+                                                   ondelete='cascade'))
+
+    _phone = db.relationship('ContactPoint', foreign_keys=phone_id,
+                             cascade="save-update, delete")
+    identifiers = db.relationship('Identifier', lazy='dynamic',
+                                  secondary="practitioner_identifiers")
+
+    def __str__(self):
+        """Print friendly format for logging, etc."""
+        return "Practitioner {0.id} {0.first_name} {0.last_name}".format(self)
+
+    @property
+    def display_name(self):
+        if self.first_name and self.last_name:
+            name = ' '.join((self.first_name, self.last_name))
+        name = name or 'Dr. {}'.format(self.last_name)
+        return escape(name)
+
+    @property
+    def phone(self):
+        if self._phone:
+            return self._phone.value
+
+    @phone.setter
+    def phone(self, val):
+        if self._phone:
+            if val:
+                self._phone.value = val
+            else:
+                self._phone = None
+        elif val:
+            self._phone = ContactPoint(
+                system='phone', use='work', value=val)
+
+    @classmethod
+    def from_fhir(cls, data):
+        practitioner = cls()
+        return practitioner.update_from_fhir(data)
+
+    def update_from_fhir(self, fhir):
+        """Update the practitioner data from the given FHIR
+
+        If a field is defined, it is the final definition for the respective
+        field, resulting in a deletion of existing values in said field
+        that are not included.
+
+        :param fhir: JSON defining portions of the user demographics to change
+
+        """
+        def v_or_n(val):
+            return val.rstrip() if val else None
+
+        if 'name' in fhir:
```

Comment:
They noticed that we didn't conform to the spec - `name` and the nested fields can be `0..*` so we're supporting accepting a list here now.  See `v_or_first()` usage in portal/models/user.py - that should also be refactored out like `v_or_n()` mentioned above.
(We have to support the list syntax as the third party FHIR lib being used packages things up by spec, so any `0..*` fields need to handle lists).

Reply:
Makes sense; updated incoming/outgoing FHIR to match that naming convention